share/translations/update.sh

```diff
@@ -1,17 +1,58 @@
-#!/bin/sh
+#!/usr/bin/env bash
+#
+# Copyright (C) 2017 KeePassXC Team <team@keepassxc.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 or (at your option)
+# version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
 
-BASEDIR=$(dirname $0)
+BASEDIR="$(dirname $0)"
 
-cd $BASEDIR/../..
+PUSH=true
+PULL=true
+UPDATE=true
 
-echo Updating source file
-lupdate-qt5 -no-ui-lines -disable-heuristic similartext -locations none -no-obsolete src -ts share/translations/keepassx_en.ts
-lupdate-qt5 -no-ui-lines -disable-heuristic similartext -locations none -pluralonly src -ts share/translations/keepassx_en_plurals.ts
+if [ "$1" == "push" ]; then
+    PULL=false
+elif [ "$1" == "pull" ]; then
+    PUSH=false
+    UPDATE=false
+elif [ "$1" == "update" ]; then
+    PUSH=false
+    PULL=false
+elif [ "$1" != "" ]; then
+    echo "Unknown command '${1}'"
+    echo "Usage: $(basename $0) [update|pull|push]"
+    exit 1
+fi
 
-echo
-echo Pushing English translation file to Transifex
-tx push -s
+cd "${BASEDIR}/../.."
 
-echo
-echo Pulling translations from Transifex
-tx pull -af --minimum-perc=40
+if $UPDATE; then
+    echo "Updating source files..."
+    lupdate-qt5 -no-ui-lines -disable-heuristic similartext -locations none -no-obsolete src -ts share/translations/keepassx_en.ts
```

Comment:
Sometimes lupdate-qt5 is actually in the system as simply lupdate. Perhaps the script can detect between the two and smartly choose the right one.